### PR TITLE
gometalinter:  use --vendor to skip ./vendor/

### PR DESCRIPTION
**Pull request by dajohi**
While here, readd gosimple and unconvert

### CODE REVIEW

**Comment by davecgh** [comment]
This fails on Go 1.7.x.

```
# github.com/decred/dcrd
./rpcserver.go:3968: cannot convert vote (type blockchain.VoteVersionTuple) to type dcrjson.VersionBits
./server.go:1264: undefined: time.Until
```

File: server.go [hunk 1]
```diff
@@ -1261,7 +1261,7 @@ func (s *server) handleAddPeerMsg(state *peerState, sp *serverPeer) bool {
 	if banEnd, ok := state.banned[host]; ok {
 		if time.Now().Before(banEnd) {
 			srvrLog.Debugf("Peer %s is banned for another %v - disconnecting",
-				host, banEnd.Sub(time.Now()))
+				host, time.Until(banEnd))
```
Comment (davecgh) [inline server.go]: This doesn't exist in Go 1.7.5 which we still support, so this can't go in.

File: rpcserver.go [hunk 2]
```diff
@@ -3965,8 +3965,7 @@ func handleGetStakeVersions(s *rpcServer, cmd interface{}, closeChan <-chan stru
 		}
 		for _, vote := range v.Votes {
 			nsv.Votes = append(nsv.Votes,
-				dcrjson.VersionBits{Version: vote.Version,
-					Bits: vote.Bits})
+				dcrjson.VersionBits(vote))
```
Comment (davecgh) [inline rpcserver.go]: This conversion is not supported by Go 1.7.x which we still support, so this can't go in.

**Comment by dajohi** [comment]
updated with gosimple removed.

**Pull request merged**